lib/galaxy/tools/__init__.py

```diff
@@ -2772,10 +2773,8 @@ def json_fix( val ):
         return [ json_fix( v ) for v in val ]
     elif isinstance( val, dict ):
         return dict( [ ( json_fix( k ), json_fix( v ) ) for ( k, v ) in val.iteritems() ] )
-    elif isinstance( val, unicode ):
-        return val.encode( "utf8" )
     else:
-        return val
+        return unicodify(val, "utf8" )
```

Comment:
The original code was changing unicode to binary string, now it's doing the opposite! I think this should be:
```
    elif isinstance( val, text_type ):
        return val.encode( "utf8" )
    else:
        return val
```